tests/Unit/Helpers/Tests/Test_MakeRandomVectorInMagnitudeRange.cpp

```diff
@@ -0,0 +1,149 @@
+    const DataType& used_for_size) noexcept {
+  std::uniform_real_distribution<> distribution(-0.05, 0.05);
+  auto spatial_metric = make_with_random_values<tnsr::ii<DataType, Dim, Fr>>(
+      generator, make_not_null(&distribution), used_for_size);
+  for (size_t d = 0; d < Dim; ++d) {
+    spatial_metric.get(d, d) += 1.0;
+  }
+  return spatial_metric;
+}
+
+void check_random_values(
+    const gsl::not_null<std::unordered_set<double>*> values,
+    const gsl::not_null<size_t*> counter, const double v,
+    const double lower_bound, const double upper_bound) noexcept {
+  CHECK(v >= lower_bound);
+  CHECK(v <= upper_bound);
+  CHECK(values->insert(v).second);
+  ++(*counter);
+}
+
+// clang-tidy is wrong, this is a function definition
+template <typename T>
+void check_random_values(
+    const gsl::not_null<std::unordered_set<double>*> values,        // NOLINT
+    const gsl::not_null<size_t*> counter, const T& c,               // NOLINT
+    const double lower_bound, const double upper_bound) noexcept {  // NOLINT
+  for (const auto& v : c) {
+    check_random_values(values, counter, v, lower_bound, upper_bound);
+  }
+}
+
+template <typename... Tags>
+void check_random_values(
+    const gsl::not_null<std::unordered_set<double>*> values,
+    const gsl::not_null<size_t*> counter,
+    const Variables<tmpl::list<Tags...>>& v, const double lower_bound,
+    const double upper_bound) noexcept {
+  expand_pack((check_random_values<decltype(get<Tags>(v))>(
+                   values, counter, get<Tags>(v), lower_bound, upper_bound),
+               cpp17::void_type{})...);
+}
+
+// Compute Magnitude
+// These functions are a workaround using Requires to
+// figure out if the inverse metric is needed
+template <typename DataType, size_t Dim, UpLo Ul, typename Fr,
+          Requires<(Ul == UpLo::Up)> = nullptr>
+Scalar<DataType> magnitude_auto_invert(
+    const Tensor<DataType, Symmetry<1>, index_list<SpatialIndex<Dim, Ul, Fr>>>&
+        vector,
+    const tnsr::ii<DataType, Dim, Fr>& metric) {
+  return magnitude(vector, metric);
+}
+
+template <typename DataType, size_t Dim, UpLo Ul, typename Fr,
+          Requires<(Ul == UpLo::Lo)> = nullptr>
+Scalar<DataType> magnitude_auto_invert(
+    const Tensor<DataType, Symmetry<1>, index_list<SpatialIndex<Dim, Ul, Fr>>>&
+        vector,
+    const tnsr::ii<DataType, Dim, Fr>& metric) {
+  const tnsr::II<DataType, Dim, Fr> inverse_metric =
+      determinant_and_inverse(metric).second;
+
+  return magnitude(vector, inverse_metric);
+}
+
+// Test copied from Test_RandomUnitNormal
+template <size_t Dim, UpLo Ul, typename DataType>
+void test_range(const gsl::not_null<std::mt19937*> generator,
+                const DataType& used_for_size) noexcept {
+  using Fr = Frame::Inertial;
+
+  std::uniform_real_distribution<> distribution1(0.0, 99.0);
+  const double r1 = distribution1(*generator);
+  std::uniform_real_distribution<> distribution2(r1, 100.0);
+  const double r2 = distribution2(*generator);
```

Comment:
You should also check any edge cases you care about, such as `r1 = 0`, `r1 = r2`, or `r1 = r2 = 0`.  (I don't know which of those are considered valid.)

Reply:
All of them are valid. I added tests for the two last edge cases you suggested.

Reply:
I removed testing the `counter` variable, because in the edge case of `r1 = r2` the duplicate entries would no be inserted into the `values` list.
I think it wasn't really testing anything relevant anyways.